front/src/modules/simulationResult/components/SpeedSpaceChart/SpeedSpaceChartV2.tsx

```diff
@@ -0,0 +1,333 @@
+  drawSpeedSpaceTrain,
+} from 'modules/simulationResult/components/SpeedSpaceChart/d3Helpers';
+import { CHART_AXES, type ChartAxes } from 'modules/simulationResult/consts';
+import type { PositionScaleDomain } from 'modules/simulationResult/types';
+import { updateSpeedSpaceSettings } from 'reducers/osrdsimulation/actions';
+import { getIsPlaying, getSpeedSpaceSettings } from 'reducers/osrdsimulation/selectors';
+import type { SpeedSpaceChart, SpeedSpaceSettingsType } from 'reducers/osrdsimulation/types';
+import { useAppDispatch } from 'store';
+import { dateIsInRange } from 'utils/date';
+
+import ElectricalProfilesLegend from './ElectricalProfilesLegend';
+import { prepareSpeedSpaceDataV2 } from './prepareData';
+import SpeedSpaceSettingsV2 from './SpeedSpaceSettingsV2';
+
+const CHART_ID = 'SpeedSpaceChart';
+const CHART_MIN_HEIGHT = 250;
+const SETTINGS_TO_AXIS = {
+  altitude: CHART_AXES.SPACE_HEIGHT,
+  curves: CHART_AXES.SPACE_RADIUS,
+  slopes: CHART_AXES.SPACE_GRADIENT,
+};
+
+export type SpeedSpaceChartV2Props = {
+  initialHeight: number;
+  onSetChartBaseHeight: (chartBaseHeight: number) => void;
+  trainSimulation: SimulationResponse;
+  selectedTrainPowerRestrictions: SimulationPowerRestrictionRange[];
+  pathProperties: PathPropertiesFormatted;
+  trainRollingStock?: LightRollingStock;
+  sharedXScaleDomain?: PositionScaleDomain;
+  setSharedXScaleDomain?: React.Dispatch<React.SetStateAction<PositionScaleDomain>>;
+};
+
+/**
+ * A chart to see the evolution of speed of one train on its journey
+ *
+ * Features:
+ * - One train only (current selected)
+ * - Vertical line to the current position
+ * - 2 marchs displayed: base and alternative
+ *
+ */
+const SpeedSpaceChartV2 = ({
+  initialHeight,
+  onSetChartBaseHeight,
+  trainSimulation,
+  selectedTrainPowerRestrictions,
+  pathProperties,
+  trainRollingStock,
+  sharedXScaleDomain,
+  setSharedXScaleDomain,
+}: SpeedSpaceChartV2Props) => {
+  const simulationIsPlaying = useSelector(getIsPlaying);
+  const speedSpaceSettings = useSelector(getSpeedSpaceSettings);
+
+  const [chart, setChart] = useState<SpeedSpaceChart | undefined>(undefined);
+  const [chartBaseHeight, setChartBaseHeight] = useState(initialHeight);
+  const [chartHeight, setChartHeight] = useState(initialHeight);
+  const [isActive, setIsActive] = useState(false);
+  const [localSettings, setLocalSettings] = useState(speedSpaceSettings);
+  const [resetChart, setResetChart] = useState(false);
+  const [showSettings, setShowSettings] = useState(false);
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation('simulation');
+
+  const toggleSetting = (settings: SpeedSpaceSettingsType) => {
+    dispatch(updateSpeedSpaceSettings(settings));
+  };
+
+  const onLocalSetSettings = (settings: SpeedSpaceSettingsType) => {
+    setLocalSettings(settings);
+    toggleSetting(settings);
+  };
+
+  const formattedTrainSimulation = useMemo(
```

Comment:
I think I prefer rawTrainSimulation and trainSimulation in the file (instead of trainSimulation in the inputs and formattedTrainSimulation in the file)
```suggestion
  const trainSimulation = useMemo(
```

Reply:
Not sure what you mean by that. You want be to change also the param name in prepareSpeedSpaceDataV2 ?